libFMS_mod/libFMS.F90

```diff
@@ -0,0 +1,471 @@
+                           diag_data_out, write_static, check_duplicate_output_fields, &
+                           get_date_dif, get_subfield_vert_size, sync_file_times, &
+                           prepend_attribute, attribute_init, diag_util_init
+  use diag_output_mod, only: diag_output_init, write_axis_meta_data, write_field_meta_data, &
+                           done_meta_data, get_diag_global_att, &
+                           set_diag_global_att, diag_field_write, diag_write_time
+  use diag_axis_mod, only: diag_axis_init, get_diag_axis, get_domain1d, &
+                           get_domain2d, get_axis_length, get_axis_global_length, &
+                           diag_subaxes_init, get_diag_axis_cart, get_diag_axis_data, &
+                           max_axes, get_axis_aux, get_tile_count, get_axes_shift, &
+                           get_diag_axis_name, get_axis_num, get_diag_axis_domain_name, &
+                           diag_axis_add_attribute, get_domainUG, axis_compatible_check, &
+                           axis_is_compressed, get_compressed_axes_ids, get_axis_reqfld, &
+                           DIAG_AXIS_NODOMAIN, DIAG_AXIS_2DDOMAIN, DIAG_AXIS_UGDOMAIN
+  use diag_data_mod, only: MAX_FIELDS_PER_FILE, DIAG_OTHER, DIAG_OCEAN, DIAG_ALL, &
+                           VERY_LARGE_FILE_FREQ, VERY_LARGE_AXIS_LENGTH, EVERY_TIME, &
+                           END_OF_RUN, DIAG_SECONDS, DIAG_MINUTES, DIAG_HOURS, DIAG_DAYS, &
+                           DIAG_MONTHS, DIAG_YEARS, MAX_SUBAXES, GLO_REG_VAL, &
+                           GLO_REG_VAL_ALT, CMOR_MISSING_VALUE, DIAG_FIELD_NOT_FOUND, &
+                           num_files, num_input_fields, num_output_fields, null_axis_id, &
+                           diag_grid, diag_fieldtype, diag_atttype, coord_type, file_type, &
+                           input_field_type, output_field_type, diag_axis_type, &
+                           diag_global_att_type, FILL_VALUE, EMPTY, &
+                           MAX_VALUE, MIN_VALUE, diag_init_time, base_time, &
+                           base_year, base_month, base_day, base_hour, base_minute, &
+                           base_second, global_descriptor
+
+  !> drifters
+  use drifters_mod, only: drifters_type, assignment(=), drifters_push, &
+                          drifters_compute_k, drifters_set_field, drifters_new, &
+                          drifters_del, drifters_set_domain, drifters_set_pe_neighbors, &
+                          drifters_set_v_axes, drifters_set_domain_bounds, & 
+                          drifters_positions2lonlat, drifters_print_checksums, &
+                          drifters_save, drifters_write_restart, drifters_distribute
+  use drifters_comm_mod, only: drifters_comm_Type, drifters_comm_new, &
+                          drifters_comm_del, drifters_comm_set_pe_neighbors, &
+                          drifters_comm_set_domain, drifters_comm_update, &
+                          drifters_comm_gather
```

Comment:
Need to understand if these are used externally or have a valid external use before exposing them to the public via this interface.

Reply:
Removed this with c267040, since it looks like it's only meant to be used in drifters_mod.

Should the other drifters_*_mod modules be removed as well?

Looking over it again it seems the drifters_*_types from the other modules are encapsulated within drifter_type so those other modules aren't necessary for using drifters_mod. The only issue is drifters_type has all those types public so they can still be accessed, although I'm not sure if that's intentional.